lib/support/modules/PesterUtil/PesterUtil.psm1

```diff
@@ -0,0 +1,156 @@
+                return
+            }
+            else {
+                Write-Host "Module $($Module.Name) found with version '$($ManifestInfo.ModuleVersion)', expecting '$ModuleVersionNoPreRelease'."
+            }
+        }
+        else {
+            # if incorrect, remove it before install
+            Remove-Item -Recurse -Force -Path $ModuleFolder
+        }
+    }
+    elseif ($PSVersionTable.PSVersion.Major -gt 5) {
+        # skip if the version already exists or if force is enabled
+        $ModuleVersionNoPreRelease = $Module.Version -replace '-.*$'
+        $ModuleFolder  = Join-Path -Path $ModuleFolder -ChildPath $ModuleVersionNoPreRelease
+        if (-not $Module.Force -and (Test-Path -Path $ModuleFolder)) {
+            Write-Verbose -Message "Module already installed."
+            return
+        }
+    }
+
+    if (-not (Test-Path $ModuleFolder)) {
+        $null = New-Item -Path $ModuleFolder -force -ItemType Directory
+    }
+
+    $urlSuffix = "/package/$($Module.Name)/$($Module.Version)".TrimEnd('/')
+    $nupkgUrl = $GalleryUrl.TrimEnd('/') + '/' + $urlSuffix.Trim('/')
+    $webclient = New-Object 'system.net.webclient'
+
+    if ($env:HTTP_PROXY){
+        if ($env:NO_PROXY){
+            Write-Host "Creating WebProxy with 'HTTP_PROXY' and 'NO_PROXY' environment variables."
+            $webproxy = New-Object -TypeName System.Net.WebProxy -ArgumentList $env:HTTP_PROXY, $true, $env:NO_PROXY
+        }
+        else {
+            Write-Host "Creating WebProxy with 'HTTP_PROXY' environment variable."
+            $webproxy = New-Object -TypeName System.Net.WebProxy -ArgumentList $env:HTTP_PROXY
+        }
+
+        $webclient.Proxy = $webproxy
+    }
+
+    Write-Verbose -Message "Downloading Package from $nupkgUrl"
+    try {
+        if (test-path $downloadedZip) {
+            Remove-Item -Force -ErrorAction SilentlyContinue -Path $downloadedZip
+        }
+        $webclient.DownloadFile($nupkgUrl, $downloadedZip)
+    }
+    catch {
+        Write-Error "Error trying to download nupkg '$nupkgUrl' to '$downloadedZip'."
+        throw $_
+    }
+    
+    if (-not (Test-Path -Path $downloadedZip)) {
+        Throw "Error trying to download nupkg '$nupkgUrl' to '$downloadedZip'."
+    }
+    
+    # Test to see if Expand-Archive is available first
+    if (Get-Command Expand-Archive) {
+        Expand-Archive -Path $downloadedZip -DestinationPath $ModuleFolder -Force
+    }
+    else {
+        # Fall back to COM object for Shell.Application zip extraction
+        Write-Host "Creating COM object for zip file '$downloadedZip'."
+        $shellcom = New-Object -ComObject Shell.Application
+        $zipcomobject = $shellcom.Namespace($downloadedZip)
+        $destination = $shellcom.Namespace($ModuleFolder)
+        $destination.CopyHere($zipcomobject.Items(), 0x610)
+        Write-Host "Nupkg installed at $ModuleFolder"
+    }
+
+    [GC]::Collect()
```

Comment:
Should probably have this in a try/finally, wouldn't wanna leave COM stuff around if an error gets thrown.

Does that COM object have a Dispose() method we should be using?

Reply:
The whole process is terminated shortly after bootstrap, and that's almost exclusively for WMF4.
That's also tested code I used for a while, I'd rather not potentially introduce more complexity.